### PR TITLE
Adding Secrets Manager Lambda Extension pattern

### PR DESCRIPTION
**Pull request by timpugh**
*Description of changes:*

The new AWS Secrets Manager Lambda Extension is now generally available. This pattern demonstrates an example written in Python of its usage.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by timpugh** [comment]
Let me know when you're ready, I got a Node.js version completed as well (figured I'd ship that over as a separate PR).

File: lambda-secrets-manager-extension-python/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    ```
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```
+    cd lambda-secrets-manager-extension-python
+    ```
+3. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam deploy --guided
```
Comment (julianwood) [inline lambda-secrets-manager-extension-python/README.md:28]: Please add a SAM build step.
1. From the command line, use AWS SAM to build the serverless application with its dependencies
    ```
    sam build

File: lambda-secrets-manager-extension-python/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+1. On line 3 of app.py we import the requests module which is used to make HTTP GET requests to the Secrets Manager extension endpoint.
+
+2. On lines 5-10 we obtain the secret name, secrets extension http port, and aws session token from environment variables and craft our request.
+
+3. On line 14 we make a request to the endpoint
+
+4. On line 16 we use json.loads to parse our secret from the response and also decode special characters. 
+
+5. On line 18 we print the secret for demonstration purposes. Please remove this in production!
+
```
Comment (julianwood) [inline lambda-secrets-manager-extension-python/README.md:50]: Suggest a testing step:
## Testing

Run the following Lambda CLI invoke command to invoke the function. Edit the {HelloWorldFunction} placeholder with the ARN of the deployed Lambda function. This is provided in the stack outputs. 
View the secret in the function output which is stored in `response.json`.

```bash
aws lambda invoke --function-name {GetSecretFunction} --cli-binary-format raw-in-base64-out response.json
cat response.json
```
----

File: lambda-secrets-manager-extension-python/template.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        - x86_64
+      Layers:
+        - arn:aws:lambda:us-east-1:177933569100:layer:AWS-Parameters-and-Secrets-Lambda-Extension:2 #This layer is specific to us-east-1. Layers for other regions can be found here: https://docs.aws.amazon.com/secretsmanager/latest/userguide/retrieving-secrets_lambda.html#retrieving-secrets_lambda_ARNs
+      Environment:
+        Variables:
+          SECRET_NAME: !Ref SecretName
+          PARAMETERS_SECRETS_EXTENSION_HTTP_PORT: "2773" #2773 is the default, but we go ahead and explicility set it to the default value anyways
+      Policies:
+        - AWSSecretsManagerGetSecretValuePolicy:
+            SecretArn: !Ref MySecret
```
Comment (julianwood) [inline lambda-secrets-manager-extension-python/template.yaml:41]: Suggest adding an Output section to more simply test:
Outputs:
  MySecret:
    Value: !Ref MySecret
    Description: MySecret ARN
  MySecretName:
    Value: !Ref SecretName
    Description: SecretName
  HelloWorldFunction:
    Value: !Ref HelloWorldFunction
    Description: HelloWorldFunction Lambda Function

File: lambda-secrets-manager-extension-python/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+## Cleanup
+
+1. Delete the stack
+    ```bash
+    aws cloudformation delete-stack --stack-name STACK_NAME
+    ```
+2. Confirm the stack has been deleted
+    ```bash
+    aws cloudformation list-stacks --query
+    ```
```
Comment (julianwood) [inline lambda-secrets-manager-extension-python/README.md:60]: Best to use sam delete:
```
1. Delete the stack, Enter `Y` to confirm deleting the stack and folder.
    ```
    sam delete
    ```

```

**Comment by julianwood** [comment]
Thanks for your contribution

**Pull request merged**